extensions/content-manager/admin/src/components/SettingsViewWrapper/index.js

```diff
@@ -0,0 +1,331 @@
+    return [
+      {
+        color: 'cancel',
+        onClick: toggleWarningCancel,
+        label: formatMessage({
+          id: `${pluginId}.popUpWarning.button.cancel`,
+        }),
+        type: 'button',
+        disabled: isEqual(modifiedData, initialData),
+        style: {
+          fontWeight: 600,
+          paddingLeft: 15,
+          paddingRight: 15,
+        },
+      },
+      {
+        color: 'success',
+        label: formatMessage({
+          id: `${pluginId}.containers.Edit.submit`,
+        }),
+        type: 'submit',
+        disabled: isEqual(modifiedData, initialData),
+        style: {
+          minWidth: 150,
+          fontWeight: 600,
+        },
+      },
+    ];
+  };
+
+  const headerProps = {
+    actions: getPluginHeaderActions(),
+    title: {
+      label: formatMessage(
+        {
+          id: `${pluginId}.components.SettingsViewWrapper.pluginHeader.title`,
+        },
+        { name: upperFirst(name) }
+      ),
+    },
+    content: formatMessage({
+      id: `${pluginId}.components.SettingsViewWrapper.pluginHeader.description.${
+        isEditSettings ? 'edit' : 'list'
+      }-settings`,
+    }),
+  };
+
+  const getRelationNameOptions = () => {
+    const attributes = getAttributes;
+    const {settings: {relationName: relationName}} = initialData;
+    const options = [];
+    Object.entries(attributes).forEach(([key, value]) => {
+     const { type, relationType } = value;
+      if(type === 'relation' && relationType === 'oneWay' ) {
+        options.push(key);
+      }
+    });
+    if(!attributes[relationName]) {
+      return ['', ...options];
+    }
+    return options;
+  }
+
+  const getSelectOptions = input => {
+    if (input.name === 'settings.defaultSortBy' || input.name === 'settings.relationKey') {
+      return [
+        'id',
+        ...getListDisplayedFields().filter(
+          name =>
+            get(getAttributes, [name, 'type'], '') !== 'media' &&
+            name !== 'id' &&
+            get(getAttributes, [name, 'type'], '') !== 'richtext'
+        ),
+      ];
+    }
+
+    if (input.name === 'settings.mainField') {
+      const attributes = getAttributes;
+      const options = Object.keys(attributes).filter(attr => {
+        const type = get(attributes, [attr, 'type'], '');
+
+        return (
+          ![
+            'json',
+            'text',
+            'relation',
+            'component',
+            'boolean',
+            'date',
+            'media',
+            'richtext',
+          ].includes(type) && !!type
+        );
+      });
+      return options;
+    }
+
+    if (input.name === 'settings.relationName') {
+       return getRelationNameOptions()
+
+    }
+
+    return input.options;
+  };
+
+  const handleSubmit = e => {
+    e.preventDefault();
+    toggleWarningSubmit();
+    emitEvent('willSaveContentTypeLayout');
+  };
+
+  if (isLoading) {
+    return <LoadingIndicatorPage />;
+  }
+
+  const isRelationSwitchable = () =>{
```

Comment:
weird logic should be summarize in comment

Reply:
done